h2o-algos/src/main/java/hex/tree/SharedTree.java

```diff
@@ -156,7 +156,7 @@ protected void ignoreInvalidColumns(int npredictors, boolean expensive) {
       error("_col_sample_rate_change_per_level", "col_sample_rate_change_per_level must be > 0" +
               " and <= 2");
     if (_train != null) {
-      double sumWeights = _train.numRows() * (hasWeightCol() ? _train.vec(_parms._weights_column).mean() : 1);
+      double sumWeights = _train.numRows() * (hasWeightCol() && _weights != null ? _weights.mean() : 1);
```

Comment:
Do we need to check for hasWeightCol? IMHO here just need to check `_weight != null` - the other check is redundant

Reply:
Fixed.